fix: request token balance for overridden tokens

stefandunca opened the pull request:
## Updates [#9091](https://github.com/status-im/status-desktop/issues/9091)

Fix: balance for tokens overrides are not requested in case of fallback when Etherscan bulk fetch is not working.

stefandunca committed:
fix: request token balance for overridden tokens

Fix balance for tokens overrides are not requested in case of fall back
when etherscan bulk fetch not working.

Updates status-desktop: #9091

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://status.im/docs/))?
- [ ] Have you tested changes with mobile?
- [ ] Have you tested changes with desktop?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (8)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | c9c10bf7 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3168/1/) | 2023-02-06 17:14:32 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230206-171305-c9c10b-pr3168.zip) |
| :heavy_check_mark: | c9c10bf7 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3168/1/) | 2023-02-06 17:17:30 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230206-171305-c9c10b-pr3168.aar) |
| :heavy_check_mark: | c9c10bf7 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3168/1/) | 2023-02-06 17:18:15 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230206-171305-c9c10b-pr3168.zip) |
| :heavy_check_mark: | c9c10bf7 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3168/1/) | 2023-02-06 17:22:39 | ~9 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3168/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | ce435850 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3168/2/) | 2023-02-07 10:31:15 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230207-102943-ce4358-pr3168.zip) |
| :heavy_check_mark: | ce435850 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3168/2/) | 2023-02-07 10:33:26 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230207-102943-ce4358-pr3168.zip) |
| :heavy_check_mark: | ce435850 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3168/2/) | 2023-02-07 10:34:30 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230207-102943-ce4358-pr3168.aar) |
| :heavy_multiplication_x: | ce435850 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3168/2/) | 2023-02-07 10:52:39 | ~22 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3168/2//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | d57a172d | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3168/3/) | 2023-02-07 13:47:54 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230207-134644-d57a17-pr3168.zip) |
| :heavy_check_mark: | d57a172d | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3168/3/) | 2023-02-07 13:50:55 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230207-134640-d57a17-pr3168.aar) |
| :heavy_check_mark: | d57a172d | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3168/3/) | 2023-02-07 13:55:43 | ~8 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230207-134644-d57a17-pr3168.zip) |
| :heavy_check_mark: | d57a172d | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3168/3/) | 2023-02-07 13:58:16 | ~11 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3168/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | a5ce4615 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3168/4/) | 2023-02-07 13:53:05 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230207-135124-a5ce46-pr3168.zip) |
| :heavy_check_mark: | a5ce4615 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3168/4/) | 2023-02-07 13:55:05 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230207-135120-a5ce46-pr3168.aar) |
| :heavy_check_mark: | a5ce4615 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3168/4/) | 2023-02-07 14:01:18 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230207-135549-a5ce46-pr3168.zip) |
| :heavy_check_mark: | a5ce4615 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3168/4/) | 2023-02-07 14:06:11 | ~7 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3168/4//consoleText) |

alaibe reviewed (commented):
What I am curious of is how is it possible that we try to fetch balance for the non overidden address. As the addresses all comes from the store they should also be using this address and not the real one?

stefandunca commented:
> What I am curious of is how is it possible that we try to fetch balance for the non overidden address. As the addresses all comes from the store they should also be using this address and not the real one?

Good question, I thought that we use a copy of the store, however, following your inquiry I realize we modify the global `tokenMap` and indeed the values don't change (logging before and after). See the updated code.

I extracted the code and added a test which passes with test values. Now I'm following this lead and look into golang global memory particularities. Thanks for the heads up.

stefandunca commented:
Found the root cause and removed the workaround. We were replacing the address but not the lookup address. The fix was to add a new entry with the overridden address and remove the original one. @alaibe please have another look.
